x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ManageOwnApiKeyClusterPrivilege.java

```diff
@@ -78,22 +78,22 @@ private boolean checkIfUserIsOwnerOfApiKeys(Authentication authentication, Strin
                  * TODO bizybot we need to think on how we can propagate appropriate error message to the end user when username, realm name
                  *   is missing. This is similar to the problem of propagating right error messages in case of access denied.
                  */
-                if (authentication.getAuthenticatedBy().getType().equals(API_KEY_REALM_TYPE)) {
+                if (authentication.getSourceRealm().getType().equals(API_KEY_REALM_TYPE)) {
```

Comment:
This is not strictly necessary since API key cannot impersonate another API key. But changed anyhow for consistency.

Reply:
I don't take issue with your change, but `ApiKeyService#getCreatorRealmType` uses the `getAuthenticatedBy` realm, so the consistency argument is weak.

Deep inside I prefer:
```
if (Authentication.AuthenticationType.API_KEY == authentication.getAuthenticationType()) { 
```
and that `ApiKeyService#getCreatorRealmType/Name` be refactored under `Authentication` .

But this is out of scope for this PR.

Reply:
I mean consistency within the same class, i.e. [the following line in the same file](https://github.com/elastic/elasticsearch/pull/52178/files/e264109c4a613d0d1e7fddd8213b4faac7f32760#diff-6ce7507e9e52811eda75d2eaefe1bb1eR88).

I agree with your analysis. The logic in `ApiKeyService#getCreatorRealmXxx` is inconsistent. Since API key cannot impersonate another API key, it works with both authenticatedBy and lookedUpBy. But I think it's better to keep them consistent. Also worthwhile to change the authentication type test as you suggested. I'll create a new issue for these changes. Thanks